mantle/kola/tests/misc/chronydhcp.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2020 Red Hat
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package misc
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/coreos/mantle/kola"
+	"github.com/coreos/mantle/kola/cluster"
+	"github.com/coreos/mantle/kola/register"
+	"github.com/coreos/mantle/platform/conf"
+	"github.com/coreos/mantle/util"
+	"github.com/vincent-petithory/dataurl"
+)
+
+var script = []byte(`#!/bin/bash
+# This script creates two veth interfaces i.e. one for the host machine 
+# and other for the container(dnsmasq server). This setup will be helpful
+# to verify the DHCP propagation of NTP servers
+# https://github.com/coreos/fedora-coreos-config/pull/412
+
+# This is needed to run a container with systemd
+setsebool container_manage_cgroup 1
+
+# create a network namespace
+ip netns add container
+
+# create veth pair and assign a namespace to veth-container
+ip link add veth-host type veth peer name veth-container
+ip link set veth-container netns container
+
+# assign an IP address to the 'veth-container' interface and bring it up
+ip netns exec container ip address add 172.16.0.1/24 dev veth-container
+ip netns exec container ip link set veth-container up
+
+# run podman commands to set up dnsmasq server
+cd /tmp
+cat <<EOF >Dockerfile
+FROM registry.fedoraproject.org/fedora:32
+RUN dnf -y install systemd dnsmasq iproute iputils
+RUN dnf clean all
+RUN systemctl enable dnsmasq
+CMD [ "/sbin/init" ]
+EOF
+podman build -t dnsmasq .
+podman run -d --rm --name dnsmasq --privileged --network ns:/var/run/netns/container dnsmasq
+podman exec -it dnsmasq /bin/bash -c "echo -e 'dhcp-range=172.16.0.10,172.16.0.20,12h\nbind-interfaces\ninterface=veth-container\ndhcp-option=option:ntp-server,129.6.15.30' >> /etc/dnsmasq.d/dhcp; sudo systemctl restart dnsmasq.service"
```

Comment:
now that we're building the container here we can just create /etc/dnsmasq.d/dhcp in the Dockerfile step.

Reply:
yeah, it makes sense to me.

Reply:
I think we still need to do `sudo systemctl restart dnsmasq.server`

Reply:
you shouldn't have to :) - but I think you already found that out